lib/linter.js

```diff
@@ -15,20 +13,46 @@ const ensureRule = (context, rule, shouldAssertion, results) => {
         if (!error.name || error.name !== "AssertionError") throw error;
 
         const pointer = (context && context.length > 0 ? context[context.length-1] : null);
-        const result = { pointer, rule, error };
-        results.push(result);
+        return { pointer, rule, error };
     }
 }
 
-const setRules = rules => {
-    activeRules = rules;
+let activeRules = {};
+
+const createNewRules = rules => {
+    rules.forEach(rule => createNewRule(rule));
+};
+
+const createNewRule = rule => {
+    if (!rule.enabled) return;
+    if (!Array.isArray(rule.object)) rule.object = [rule.object];
+    if (rule.alphabetical && rule.alphabetical.properties && !Array.isArray(rule.alphabetical.properties)) {
+        rule.alphabetical.properties = [rule.alphabetical.properties];
+    }
+    if (rule.truthy && !Array.isArray(rule.truthy)) rule.truthy = [rule.truthy];
+    activeRules[rule.name] = rule;
 }
 
-const lint = (objectName, object, options = {}) => {
+const resetRules = () => {
+    activeRules = {};
+};
+
+const lint = async (objectName, object, options = {}) => {
+
+    console.log('now linting', object);
+    console.log('activeRules', activeRules);
```

Comment:
Problem 1: This is outputting empty in tests using this loadRuleFiles logic.

http://github.com/wework/speccy/blob/8e94f39bb4caa49ae6d3cf3f43fb408a4978f31c/test/linter.test.js#L21